lua/bufmode.lua

```diff
@@ -15,68 +13,111 @@ local keymaps =
 	['w'] = 'bnext',
 }
 
--- Add mappings for `barbar.nvim`
-if vim.fn.exists ':BufferClose' > 0 then
-	keymaps = vim.tbl_extend('force', keymaps,
-	{
-		['$'] = 'BufferLast',
-		['0'] = 'BufferFirst',
-		['B'] = 'BufferMovePrevious',
-		['b'] = 'BufferPrevious',
-		['d'] = 'BufferDelete',
-		['p'] = 'BufferPick',
-		['r'] = 'BufferClose',
-		['W'] = 'BufferMoveNext',
-		['w'] = 'BufferNext',
-	})
-end
+local barbar_keymaps =
+{
+  ['$'] = 'BufferLast',
+  ['0'] = 'BufferFirst',
+  ['B'] = 'BufferMovePrevious',
+  ['b'] = 'BufferPrevious',
+  ['d'] = 'BufferDelete',
+  ['p'] = 'BufferPick',
+  ['r'] = 'BufferClose',
+  ['W'] = 'BufferMoveNext',
+  ['w'] = 'BufferNext',
+}
 
--- add user mappings
-keymaps = vim.tbl_extend('force', keymaps, vim.g.bufmode_mappings or {})
+local bufferline_keymaps =
+{
+  ['B'] = 'BufferLineMovePrev',
+  ['b'] = 'BufferLineCyclePrev',
+  ['p'] = 'BufferLinePick',
+  ['W'] = 'BufferLineMoveNext',
+  ['w'] = 'BufferLineCycleNext',
+}
 
--- create a `new` link for some `existing` mapping
-local function inherit(child, parent)
-	keymaps[child] = keymaps[parent]
-end
+---@class BufmodeOptions
+---@field barbar? boolean Whether to add bindings for barbar.nvim
+---@field bufferline? boolean Whether to add bindings for bufferline.nvim
+---@field keymaps? table custom key bindings to apply
+---@field enter_mapping? string custom binding to enter buffers mode
 
--- Turn some special character value into a character code.
-local function tochar(val)
-	return vim.api.nvim_replace_termcodes(val, true, true, true)
-end
+---@param opts? BufmodeOptions
+function M.setup(opts)
+  opts = opts or {}
+  local libmodal = require 'libmodal'
```

Comment:
You can just move the `require` to line 106, since `libmodal` is only referenced once.